lib/dep-graph-builders/pnpm/lockfile-parser/lockfile-v5.ts

```diff
@@ -0,0 +1,64 @@
+import { parse } from 'dependency-path';
+import { PnpmDeps } from '../types';
+import { PnpmLockfileParser } from './lockfile-parser';
+import { PnpmWorkspaceArgs } from '../../types';
+import { valid } from 'semver';
+import * as pathUtil from 'path';
+
+export class LockfileV5Parser extends PnpmLockfileParser {
+  public specifiers: Record<string, string>;
+
+  public constructor(rawPnpmLock: any, workspaceArgs?: PnpmWorkspaceArgs) {
+    super(rawPnpmLock, workspaceArgs);
+    const depsRoot = this.getRoot(rawPnpmLock);
+    this.specifiers = depsRoot.specifiers;
+  }
+
+  public parseDepPath(depPath: string) {
+    // The 'dependency-path' parsing package only works for lockfiles v5
+    const { name, version } = parse(depPath);
+    return {
+      name,
+      version: this.excludeTransPeerDepsVersions(version!),
```

Comment:
Is there a case where the version is not provided?
If there is, make sure to assert against undefined or null or provide a default value.